### DISCUSSION TITLE
Enable SWC Emotion importMap

### DISCUSSION THREAD

**Discussion by Zn4rK** [vercel/next.js#41732] [Ideas]
### Describe the feature you'd like to request

I recently tried to move from babel to SWC in one of my project, but noticed that I couldn't due to the fact that the SWC plugin for emotion does not have importMap support. In the current setup I have, `emotion` is re-exported and then used under the new export across the project.

importMap makes that possible:
https://github.com/emotion-js/emotion/tree/main/packages/babel-plugin#importmap

### Describe the solution you'd like

It looks like importMap support is already in the emotion plugin:

https://github.com/vercel/next.js/blob/canary/packages/next-swc/crates/emotion/src/import_map.rs
https://github.com/vercel/next.js/blob/canary/packages/next-swc/crates/emotion/tests/testImportMap.json

(Edit: These moved to [here](https://github.com/swc-project/plugins/tree/main/packages/emotion/transform/src) and [here](https://github.com/swc-project/plugins/tree/main/packages/emotion/transform/tests) after https://github.com/vercel/next.js/pull/41613 was merged. Probably should've used a permalink, but w/e).

But the options are gated in the config:
https://github.com/vercel/next.js/blob/99d88f2a585618b80058b201f765e66c88e5f5b8/packages/next/build/swc/options.js#L156

Is this an oversight or is there a specific reason for the importMap not being accepted as options to the emotion compiler settings?

### Describe alternatives you've considered

- Continue using babel.
- Using a patch to manually edit `node_modules/next/dist/build/swc/options.js` and set the importMap (which seems to work).

**Comment by Zn4rK** [comment]
@balazsorban44 I'm super confused as to why this was converted into a discussion, and that it also got the label `area: next/image`. 

The SWC emotion plugin supports [`importMap`](https://github.com/swc-project/plugins/issues/53), but the emotion compiler options for next.js does not allow you to specify the importMap. It's not much to discuss. ¯\\_(ツ)_/¯

Do you guys want a PR for this, or will this be handled in some upcoming release - should we maybe move this discussion back to an issue to make sure that it's trackable?

**Comment by balazsorban44** [comment]
Hi, I'm sorry for the incorrect labeling, it was by accident. As for converting to a discussion, every feature request is converted as stated in the issue template and in the contribution docs. https://github.com/vercel/next.js/blob/canary/contributing/repository/triaging.md

PRs are welcome! 🙏
